doc/api/module.md

```diff
@@ -317,6 +317,37 @@ isBuiltin('fs'); // true
 isBuiltin('wss'); // false
 ```
 
+### `module.mapCallSite(callSite)`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1.1 - Active development
+
+* `callSite` {Object | Array} A [CallSite][] object or an array of CallSite objects.
+* Returns: {Object | Array} The original source code location(s) for the given CallSite object(s).
+
+Reconstructs the original source code location from a [CallSite][] object through the source map.
+
+```mjs
+import { mapCallSite } from 'node:module';
+import { getCallSite } from 'node:util';
+
+mapCallSite(getCallSite()); // Reconstructs the original source code location for the whole stack
+
+mapCallSite(getCallSite()[0]); // Reconstructs the original source code location for the first frame
+```
+
+```cjs
+const { mapCallSite } = require('node:module');
+const { getCallSite } = require('node:util');
+
+mapCallSite(getCallSite()); // Reconstructs the original source code location for the whole stack
+
+mapCallSite(getCallSite()[0]); // Reconstructs the original source code location for the first frame
```

Comment:
I feel the DX of this is bad. Don't you think? 

I'm sure if users just call `getCallSite()` should be better. Internally, we could check if typescript is enabled and attempt to read source map (basically what `mapCallSites` does, but inlined into `getCallSites()`)

Reply:
We can add it inside but still needs an option to enable it, because its not only  typescript but also minified code needs sourcemap

Reply:
What if?

```cjs
try {
  await attemptToReadSourceMap()
} catch (e) {
  currentGetCallSite()
}
```

Reply:
I added a commit to improve `util.getCallSite` dx when `--enable-source-maps` flag is passed or added manually